gen3-integration-tests/services/indexd.py

```diff
@@ -92,6 +124,22 @@ def delete_files(self, guids: list, user="indexing_account"):
             except Exception as e:
                 logger.exception(msg=f"Failed to delete record with guid {guid} : {e}")
 
+    def delete_file_indices(self, records: dict):
+        for key, val in records.items():
+            try:
+                indexd_record = self.get_record(indexd_guid=val["did"])
+                indexd_rev = self.get_rev(json_data=indexd_record)
+                if indexd_rev is None:
+                    logger.info("Indexd record returned None")
+                    continue
+                logger.info(f"{val['did']} found, performing delete.")
+                self.delete_record(guid=indexd_record["did"], rev=indexd_rev)
+            except Exception as e:
+                if "404" not in f"{e}" and "did" not in f"{e}":
+                    logger.error(f"404 status code not returned. Exception : {e}")
+                    raise
```

Comment:
`delete_record` returns a status code: https://github.com/uc-cdis/gen3-code-vigil/blob/c641536fe2c5e222afb94bb7ad0fc6a929f4de2f/gen3-integration-tests/services/indexd.py#L113

so i think in case we get a 404, it wouldn't raise an exception, so you would never run this `except` block?

i think you can remove this `try/except` block and do:
```
status_code = self.delete_record(...)
if status_code!= 200 and status_code != 404:
    logger.error(...)
```

Reply:
There is a mixup of deletion functions here, we should be using sdk's deletion where possible. I think we added the method with direct api call to /delete endpoint since we noticed the sdk's deletion did not work for records created using gen3client. Not sure if that should even be the case.